spotbugs-tests/src/test/java/edu/umd/cs/findbugs/detect/RegressionIdeas20110722Test.java

```diff
@@ -11,6 +12,8 @@
 
 class RegressionIdeas20110722Test extends AbstractIntegrationTest {
 
+    // This test, or rather the detector needs to be fixed
```

Comment:
Can you describe in more detail what is broken?

Reply:
I'm still trying to fix this problem, this is the reason the PR is still a draft, and why I haven't explained it in the comment. If I give up, I definitely will provide some details in the code, or rather open an issue and refer that here.

Basically, the found buginstances should be:
- 4 RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE 
   - at getHashCode3() method line 34
   - at getHashCode4() method line 41
   - at getHashCode4() method line 48
   - at getHashCode4() method line 55
- 2 NP_PARAMETER_MUST_BE_NONNULL_BUT_MARKED_AS_NULLABLE
  - at getHashCode6() method parameter x
  - at getHashCode7() method parameter x

Instead it is the following:
- 4 NP_NULL_ON_SOME_PATH
   - at getHashCode3() method line 37
   - at getHashCode4() method line 44
   - at getHashCode5() method line 51
   - at getHashCode6() method line 58

I think the root of the problem might be in `IsNullValueAnalysis`, which I'm not so familiar with and need to understand how it works to be able to solve it.